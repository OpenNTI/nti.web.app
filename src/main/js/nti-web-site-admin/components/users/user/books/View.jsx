import React from 'react';
import PropTypes from 'prop-types';
import {Loading} from '@nti/web-commons';
import {scoped} from '@nti/lib-locale';
import {LinkTo} from '@nti/web-routing';

import ErrorMessage from '../../../common/ErrorMessage';
import Card from '../../../common/Card';

import BookListItem from './BookListItem';
import Store from './Store';

const DEFAULT_TEXT = {
	error: 'Unable to load transcript.'
};
const t = scoped('nti-site-admin.users.user.Transcript', DEFAULT_TEXT);

const propMap = {
	items: 'items',
	loading: 'loading',
	error: 'error',
};

@Store.connect(propMap)
export default class SiteAdminUserBooks extends React.Component {
	static propTypes = {
		user: PropTypes.object,
		store: PropTypes.object,

		items: PropTypes.array,
		loading: PropTypes.bool,
		error: PropTypes.any,
	}

	get store () {
		return this.props.store;
	}


	componentDidMount () {
		const {user} = this.props;

		this.store.loadBooks(user);
	}


	componentWillUnmount () {
		const {user} = this.props;

		this.store.unloadBooks(user);
	}


	componentDidUpdate (oldProps) {
		const {user:newUser} = this.props;
		const {user: oldUser} = oldProps;

		if (newUser !== oldUser) {
			this.store.loadBooks(newUser);
		}
	}


	render () {
		const {loading, error} = this.props;

		return (
			<div className="site-admin-user-transcripts">
				{loading && (<Loading.Mask />)}
				{!loading && this.renderItems()}
				{error && (<ErrorMessage>{t('error')}</ErrorMessage>)}
			</div>
		);
	}

	renderItems () {
		const {items} = this.props;

		if (!items || !items.length) { return this.renderEmptyState(); }

		return (
			<Card>
				<ul>
					{items.map((item, index) => {
						return (
							<li key={index}>
								<LinkTo.Object object={item} context="site-admin.users.user-books.list">
									<BookListItem book={item.Bundle} />
								</LinkTo.Object>
							</li>
						);
					})}
				</ul>
			</Card>
		);
	}


	renderEmptyState () {

	}
}
