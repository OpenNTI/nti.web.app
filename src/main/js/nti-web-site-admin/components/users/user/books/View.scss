@import '~@nti/style-common';

.site-admin-user-transcripts {
	min-height: 200px;
	position: relative;

	ul {
		li:not(:last-of-type) {
			box-shadow: 0 1px 0 0 var(--border-grey-light);
		}

		li a {
			text-decoration: none;
		}
	}
}
