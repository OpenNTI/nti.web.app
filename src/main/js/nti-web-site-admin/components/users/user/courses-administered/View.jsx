import { useEffect } from 'react';

import { scoped } from '@nti/lib-locale';
import { EmptyState, List, Loading } from '@nti/web-commons';
import { LinkTo } from '@nti/web-routing';
import { EnrollmentListItem } from '@nti/web-course';

import ErrorMessage from '../../../common/ErrorMessage';
import Card from '../../../common/Card';

import Store from './Store';

const t = scoped('nti-site-admin.users.user.courses-administered.View', {
	error: 'Unable to load courses.',
	noCourses: 'This user is not administering any courses',
});

function SiteAdminUserAdministeredCourses({ user }) {
	const { loading, error, load, unload } = Store.useValue();

	useEffect(() => {
		load(user);
		return () => {
			unload(user);
		};
	}, [user]);

	const onEnrollmentChange = () => {
		load(user);
	};

	return (
		<div
			css={css`
				margin-top: 54px;
				min-height: 200px;
				position: relative;
			`}
		>
			{loading ? (
				<Loading.Mask />
			) : (
				<>
					<Items {...{ user, onEnrollmentChange }} />
				</>
			)}
			{error && <ErrorMessage>{t('error')}</ErrorMessage>}
		</div>
	);
}

export default Store.compose(SiteAdminUserAdministeredCourses);

const ListItem = styled.li`
	&:not(:last-of-type) {
		box-shadow: 0 1px 0 0 var(--border-grey-light);
	}

	a {
		text-decoration: none;
	}
`;

function Items({ onEnrollmentChange }) {
	const { items } = Store.useValue();

	if (!items?.length) {
		return <EmptyState>{t('noCourses')}</EmptyState>;
	}

	return (
		<Card>
			<List.Unadorned>
				{items.map((item, index) => (
					<ListItem key={index}>
						<LinkTo.Object
							object={item}
							context="site-admin.users.user-administered-courses.list"
						>
							<EnrollmentListItem enrollment={item} />
						</LinkTo.Object>
					</ListItem>
				))}
			</List.Unadorned>
		</Card>
	);
}
