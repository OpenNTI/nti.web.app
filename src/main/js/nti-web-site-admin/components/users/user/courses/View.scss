@import '~@nti/style-common';

.site-admin-user-transcripts {
	min-height: 200px;
	position: relative;

	.user-transcript-controls {
		display: flex;
		flex-direction: row;
		justify-content: flex-end;
		margin-bottom: 1rem;
	}

	ul {
		li:not(:last-of-type) {
			box-shadow: 0 1px 0 0 var(--border-grey-light);
		}

		li a {
			text-decoration: none;
		}
	}
}
