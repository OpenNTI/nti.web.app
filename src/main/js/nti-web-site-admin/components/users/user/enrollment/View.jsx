import React from 'react';

export default function UserEnrollment () {
	return (
		<div>
			User Enrollment
		</div>
	);
}
