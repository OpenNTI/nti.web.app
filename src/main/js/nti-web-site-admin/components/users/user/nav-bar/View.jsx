
import Card from '../../../common/Card';

import Identity from './Identity';
import Tabs from './Tabs';
import EmailUpdate from './EmailUpdate';

export default function SiteAdminUserIdentity(props) {
	return (
		<>
			<Card className="site-admin-user-nav-bar">
				<Identity {...props} />
				<Tabs {...props} />
			</Card>
			<EmailUpdate {...props} />
		</>
	);
}
