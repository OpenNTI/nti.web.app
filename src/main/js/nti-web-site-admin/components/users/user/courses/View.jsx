import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';

import { List, Loading, Button } from '@nti/web-commons';
import { decorate } from '@nti/lib-commons';
import { scoped } from '@nti/lib-locale';
import { LinkTo } from '@nti/web-routing';
import { EnrollmentListItem, Enrollment } from '@nti/web-course';

import ErrorMessage from '../../../common/ErrorMessage';
import Card from '../../../common/Card';

import Store from './Store';

const DEFAULT_TEXT = {
	error: 'Unable to load courses.',
	noCourses: 'This user is not enrolled in any courses',
	manage: 'Manage Courses',
};
const t = scoped('nti-site-admin.users.user.courses.View', DEFAULT_TEXT);

const propMap = {
	items: 'items',
	loading: 'loading',
	error: 'error',
};

class SiteAdminUserCourses extends React.Component {
	static propTypes = {
		user: PropTypes.object,
		store: PropTypes.object,

		items: PropTypes.array,
		loading: PropTypes.bool,
		error: PropTypes.any,
	};

	get store() {
		return this.props.store;
	}

	componentDidMount() {
		const { user } = this.props;

		this.store.loadTranscript(user);
	}

	componentWillUnmount() {
		const { user } = this.props;

		this.store.unloadTranscript(user);
	}

	componentDidUpdate(prevProps) {
		const { user: newUser } = this.props;
		const { user: oldUser } = prevProps;

		if (newUser !== oldUser) {
			this.store.loadTranscript(newUser);
		}
	}

	onEnrollmentChange = () => {
		const { user } = this.props;

		this.store.loadTranscript(user);
	};

	render() {
		const { loading, error } = this.props;

		return (
			<div className="site-admin-user-transcripts">
				{loading && <Loading.Mask />}
				{!loading && this.renderControls()}
				{!loading && this.renderItems()}
				{error && <ErrorMessage>{t('error')}</ErrorMessage>}
			</div>
		);
	}

	renderControls() {
		const { user } = this.props;

		if (!user.hasLink('EnrollUser')) {
			return null;
		}

		return (
			<div className="user-transcript-controls">
				<Enrollment.Admin.Prompt.Trigger
					user={user}
					onChange={this.onEnrollmentChange}
				>
					<Button rounded>{t('manage')}</Button>
				</Enrollment.Admin.Prompt.Trigger>
			</div>
		);
	}

	renderItems() {
		const { items } = this.props;

		if (!items || !items.length) {
			return this.renderEmptyState();
		}

		return (
			<Card>
				<List.Unadorned>
					{items.map((item, index) => {
						return (
							<li key={index}>
								<LinkTo.Object
									object={item}
									context="site-admin.users.user-transcript.list"
								>
									<EnrollmentListItem
										enrollment={item}
										onChange={this.onEnrollmentChange}
									/>
								</LinkTo.Object>
							</li>
						);
					})}
				</List.Unadorned>
			</Card>
		);
	}

	renderEmptyState() {
		return <div className="empty-state">{t('noCourses')}</div>;
	}
}

export default decorate(SiteAdminUserCourses, [Store.connect(propMap)]);
