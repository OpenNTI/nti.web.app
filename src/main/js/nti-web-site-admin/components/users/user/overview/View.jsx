import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';
import { Widgets } from '@nti/web-reports';
import { getService } from '@nti/web-client';

import DateValue from '../../../common/DateValue';

import ActiveTimes from './ActiveTimes';
import RecentSessions from './RecentSessions';
import LastLogin from './LastLogin';

const { ActiveDays } = Widgets;

const HISTORICAL_SESSIONS_LINK = 'HistoricalSessions';

const NUM_SESSIONS_TO_SHOW = 5;

export default class SiteAdminUserOverview extends React.Component {
	static propTypes = {
		user: PropTypes.object,
	};

	constructor(props) {
		super(props);

		this.state = { loading: true };
	}

	componentDidMount() {
		this.loadData();
	}

	async loadData() {
		const { user } = this.props;

		try {
			const service = await getService();
			const historicalSessions = user.getLink(HISTORICAL_SESSIONS_LINK);
			const results = historicalSessions
				? await service.get(historicalSessions)
				: {};
			const items = (results.Items || []).slice(0, NUM_SESSIONS_TO_SHOW);

			this.setState({ loading: false, historicalSessions: items });
		} catch (e) {
			this.setState({ loading: false, historicalSessions: [] });
		}
	}

	render() {
		const { user } = this.props;
		const { loading, historicalSessions } = this.state;

		return (
			<div className="site-admin-user-overview">
				<div className="joined">
					<DateValue date={user.getCreatedTime()} label="Joined" />
				</div>
				<div className="last-login">
					<LastLogin
						user={user}
						loading={loading}
						historicalSessions={historicalSessions}
					/>
				</div>
				<div className="activity-by-day">
					<ActiveDays entity={user} />
				</div>
				<div className="active-times">
					<ActiveTimes user={user} />
				</div>
				<div className="recent-sessions">
					<RecentSessions
						user={user}
						loading={loading}
						historicalSessions={historicalSessions}
					/>
				</div>
			</div>
		);
	}
}
