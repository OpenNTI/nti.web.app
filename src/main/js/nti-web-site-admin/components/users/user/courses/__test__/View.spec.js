import { render, waitFor } from '@testing-library/react';

import View from '../View';

const getMockUser = numberOfCourses => {
	let Items = [];

	for (let i = 0; i < numberOfCourses; i++) {
		const index = i + 1;

		Items.push({
			getPresentationProperties: () => {
				return {
					label: 'Course' + index,
					title: 'course' + index,
				};
			},
			hasLink: () => true,
		});
	}

	return {
		hasLink: () => true,
		fetchLink: async () => {
			return {
				Items,
			};
		},
	};
};

/* eslint-env jest */
describe('Site admin user course list test (5 courses)', () => {
	test('Basic render test', async () => {
		const user = getMockUser(5);
		const cmp = render(<View user={user} />);

		await waitFor(() => {
			expect(
				cmp.container.querySelectorAll(
					'.nti-course-enrollment-list-item'
				).length
			).toBe(5);
		});

		expect(cmp.asFragment()).toMatchSnapshot();
	});
});

describe('Site admin user course list test (no courses)', () => {
	test('Basic render test', async () => {
		const user = getMockUser(0);

		const cmp = render(<View user={user} />);

		await waitFor(() => {
			expect(
				cmp.container.querySelectorAll('.empty-state-component').length
			).toBe(1);
		});

		expect(cmp.asFragment()).toMatchSnapshot();
	});
});
