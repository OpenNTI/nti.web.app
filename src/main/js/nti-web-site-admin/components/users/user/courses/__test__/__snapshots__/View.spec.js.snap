// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user course list test (5 courses) Basic render test 1`] = `
<DocumentFragment>
  <div
    css="__inline_css_extracted_to_hashed_class_name"
  >
    <div
      css="__inline_css_extracted_to_hashed_class_name"
    >
      <div
        class="nti-course-enrollment-admin-prompt-trigger"
      >
        <button
          class="nti-button button primary medium pv-md ph-xl roundedSmall"
          tabindex="0"
        >
          <div
            data-button-label="true"
          >
            Manage Courses
          </div>
        </button>
      </div>
    </div>
    <div
      class="site-admin-card"
      css="__inline_css_extracted_to_hashed_class_name"
    >
      <ul
        class="unadorned-list"
      >
        <li>
          <div
            class="nti-link-to-path"
            to="#"
          >
            <div
              class="nti-course-enrollment-list-item"
            >
              <img
                class="course-enrollment-icon"
                src=""
              />
              <div
                class="course-enrollment-meta"
              >
                <div
                  class="label"
                >
                  Course1
                </div>
                <div
                  class="title"
                >
                  course1
                </div>
              </div>
              <div
                class="course-enrollment-controls"
              >
                <div
                  class="nti-course-enrollment-admin-prompt-trigger"
                >
                  <i
                    class="icon-edit"
                  />
                </div>
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            class="nti-link-to-path"
            to="#"
          >
            <div
              class="nti-course-enrollment-list-item"
            >
              <img
                class="course-enrollment-icon"
                src=""
              />
              <div
                class="course-enrollment-meta"
              >
                <div
                  class="label"
                >
                  Course2
                </div>
                <div
                  class="title"
                >
                  course2
                </div>
              </div>
              <div
                class="course-enrollment-controls"
              >
                <div
                  class="nti-course-enrollment-admin-prompt-trigger"
                >
                  <i
                    class="icon-edit"
                  />
                </div>
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            class="nti-link-to-path"
            to="#"
          >
            <div
              class="nti-course-enrollment-list-item"
            >
              <img
                class="course-enrollment-icon"
                src=""
              />
              <div
                class="course-enrollment-meta"
              >
                <div
                  class="label"
                >
                  Course3
                </div>
                <div
                  class="title"
                >
                  course3
                </div>
              </div>
              <div
                class="course-enrollment-controls"
              >
                <div
                  class="nti-course-enrollment-admin-prompt-trigger"
                >
                  <i
                    class="icon-edit"
                  />
                </div>
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            class="nti-link-to-path"
            to="#"
          >
            <div
              class="nti-course-enrollment-list-item"
            >
              <img
                class="course-enrollment-icon"
                src=""
              />
              <div
                class="course-enrollment-meta"
              >
                <div
                  class="label"
                >
                  Course4
                </div>
                <div
                  class="title"
                >
                  course4
                </div>
              </div>
              <div
                class="course-enrollment-controls"
              >
                <div
                  class="nti-course-enrollment-admin-prompt-trigger"
                >
                  <i
                    class="icon-edit"
                  />
                </div>
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            class="nti-link-to-path"
            to="#"
          >
            <div
              class="nti-course-enrollment-list-item"
            >
              <img
                class="course-enrollment-icon"
                src=""
              />
              <div
                class="course-enrollment-meta"
              >
                <div
                  class="label"
                >
                  Course5
                </div>
                <div
                  class="title"
                >
                  course5
                </div>
              </div>
              <div
                class="course-enrollment-controls"
              >
                <div
                  class="nti-course-enrollment-admin-prompt-trigger"
                >
                  <i
                    class="icon-edit"
                  />
                </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Site admin user course list test (no courses) Basic render test 1`] = `
<DocumentFragment>
  <div
    css="__inline_css_extracted_to_hashed_class_name"
  >
    <div
      css="__inline_css_extracted_to_hashed_class_name"
    >
      <div
        class="nti-course-enrollment-admin-prompt-trigger"
      >
        <button
          class="nti-button button primary medium pv-md ph-xl roundedSmall"
          tabindex="0"
        >
          <div
            data-button-label="true"
          >
            Manage Courses
          </div>
        </button>
      </div>
    </div>
    <figure
      class="empty-state-component"
    >
      <h1>
        This user is not enrolled in any courses
      </h1>
    </figure>
  </div>
</DocumentFragment>
`;
